2024/XenSummit/Challenges_and_Status_of_Enabling_TrenchBoot_in_Xen_Hypervisor.md

```diff
@@ -0,0 +1,310 @@
+  boot mode
+
+---
+
+# Phase 3 - TrenchBoot boot protocol
+
+.center[.image-99[![](/img/tb_aem_phase3.png)]]
+
+???
+
+- Rebased changes onto the most recent TrenchBoot support patches to Linux
+  kernel introducing a unified TrenchBoot boot protocol
+- Final shape of patches is not yet determined, the work on finalizing the
+  boot protocol including UEFI boot mode is pending on the [Linux kernel
+  mailing list, currently patchset
+  v8](https://lore.kernel.org/lkml/8d543a15-af62-4403-b2e0-3b395edfe9e4@amd.com/T/)
+
+---
+
+# Phase 4 - AMD support
+
+.center[.image-90[![](/img/tb_aem_phase4.png)]]
+
+???
+
+- Implemented support for booting with AMD SKINIT and TPM 1.2/2.0 in legacy
+  BIOS boot mode
+- Implemented the newest TrenchBoot boot protocol in the Secure Kernel Loader
+- Simplified installation of the required packages by providing a RPM
+  repository for QubesOS
+
+---
+
+# Try TenchBoot
+
+It is relatively easy to get a hardware which supports DRTM:
+
+- Intel-based tested and known to work:
+  * Protectli VP4670
+  * HP EliteDesk 800 G2 Desktop Mini
+
+.center[.image-30[![](/img/VP4600.png)].image-30[![](/img/hp_elitedesk_800_g2.jpg)]]
+
+- AMD-based tested and known to work:
+  * Terminal HP T630
+
+.center[.image-30[![](/img/hp_t630.png)]]
+
+???
+
+These are the tested platforms that are working well. We have also faced
+issues with other platforms with BIOS bugs, which could not be workarounded,
+like Lenovo M920 or SUpermicro M11SDV
+
+Not all Intel platform support DRTM technology - Intel TXT. It requires a
+capable CPU and chipset. Typically vPro platforms satisfy these requirements,
+although it is still possible to have a Intel TXT initialization bug in BIOS,
+which prevents using DRTM unless a BIOS fix is issued.
+
+For AMD platforms, almost every CPU supports AMD SKINIT (CPU needs to supports
+SVM basically and BIOS must not disable it).
+
+---
+
+# Current ongoing efforts
+
+- Provide package repositories for other major distros (Debian, Ubuntu Fedora)
+- Prepare tools for assessing the readiness of the system and BIOS to perform
+  Dynamic Launch
+- Adding DRTM Ready indicator to the system security reports: QubesOS Security
+  Report, [fwupd HSI](https://fwupd.github.io/libfwupdplugin/hsi.html)
+
+.center[.image-40[![](/img/qubes-sec-report.png)]]
+
+---
+
+# Phase 5 - Xen as DLME for UEFI boot mode
+
+### Future planned work
+
+- Xen will be able to initiate the DL Event via the code module exposed by DCE
+  Preamble (GRUB)
+  * Although Xen does not use that much information from UEFI we want to unify
+    the flow with TrenchBoot implementation for Linux kernel
+  * Native EFI PE entry point shall be supported for UEFI Secure Boot purposes
+- Xen will need to obtain the module from Secure Launch Resource Table linked
+  in the EFI System Table Configuration Tables
+- Xen will fill the necessary information for the DCE (like Xen address in
+  memory to be measured) and call the module
```

Comment:
To the best of my knowledge, none of these are planned. I think we agreed on using plain MB2 even on UEFI.